miden-tx/src/executor/mod.rs

```diff
@@ -137,8 +138,10 @@ impl<D: DataStore> TransactionExecutor<D> {
         block_ref: u32,
         notes: &[NoteId],
         tx_script: Option<TransactionScript>,
+        note_args: Option<BTreeMap<NoteId, Word>>,
```

Comment:
Why do we pass `note_args` as `Option`? We could probably just do `BTreeMap<NoteId, Word>` and an empty map would be equivalent to no args.